contracts/Staking.sol

```diff
@@ -50,52 +60,63 @@ contract Staking {
     if you are staking for a contract, the contract must be able to call unstake
     @param amount the amount you want to stake. Can be left blank when you are staking the native currency
     @param user the user for whom you are staking
+    @param tokenAddress Token address
     @param data should be disregarded for the current deployment
      */
-    function stakeFor(uint256 amount, address user, bytes memory data) public payable {
+    function stakeFor(uint256 amount, address user, address tokenAddress, bytes memory data) public payable {
         // disregard passed-in amount
-        if(contractUsesNativeToken()) {
+        if(_isNativeToken(tokenAddress)) {
             amount = msg.value;
+            tokenAddress = address(0);
```

Comment:
I think we should also allow to configure if the Native token is accepted. Here it is assumed you can always stake in RBTC. The problem is we cannot use the Whitelist so we would need to create a flag, and a setter function like we did [here](https://github.com/rsksmart/rif-marketplace-nfts/blob/8be8c5ad0f5c49c9f619dd47d31c5864fc03d986/contracts/ERC721SimplePlacementsV1.sol#L43). We can name it `isNativeTokenAllowed` or something on those lines.